doc/adr/2023-10-18 subdomain routing with cert per unit.md

```diff
@@ -0,0 +1,110 @@
+# Obtain a separate cert per every unit when using subdomain routing
+
+## Context and Problem Statement
+Until now, the TLS cert solution was designed with only path routing in mind.
+When using Traefik in the subdomain routing mode, all Ingress will use
+Traefik's default certificate, because charm code only obtains one cert (for
+the external hostname), and traefik obviously cannot match that cert with
+subdomains.
+
+https://github.com/canonical/traefik-k8s-operator/issues/244
+
+## Considered Options
+
+- One cert per app/unit (no wildcards)
+- Only one cert (as many wildcard SANs as necessary)
+- Only one cert (keep appending subdomains; no wildcards)
+- One cert per app (wildcard SANs only for ingress-per-unit)
+- One cert per app (keep appending IPU subdomains; no wildcards)
+
+## Decision Outcome
+
+Chosen option: "One cert per app/unit (no wildcards)", because we must support
+the use-case of strict subdomain routing (no wildcards).
+
+### Consequences
+
+* Good, because no wildcard certs (more secure).
+* Good, because not limited by SAN count per cert.
+* Bad, because may get rate limited by the CA.
+  * IPA: new CSR (CRR) for every added (removed) app
+  * IPU: new CSR (CRR) for every added (removed) unit
+
+## Pros and Cons of the Options
+
+### One cert per app/unit (no wildcards)
+```
+IPA: model-app.example.com
+IPU: model-app-0.example.com
+
+SAN example: model-app-0.example.com
+```
+
+* Good, because no wildcard certs (more secure).
+* Good, because not limited by SAN count per cert.
+* Bad, because may get rate limited by the CA.
+
+### Only one cert (as many wildcard SANs as necessary)
+```
+IPA: model-app.example.com
+IPU: model-app-0.example.com
+
+SAN example: *.example.com
+```
+
+* Good, because won't get rate limited by the CA.
+* Good, because simple to implement.
+* Bad, because the number of SANs per cert is limited.
+* Bad, because a wildcard cert is too permissive.
+
+### Only one cert (keep appending subdomains; no wildcards)
+```
+IPA: model-app.example.com
+IPU: model-app-0.example.com
+
+SAN example: app1.example.com, app2.example.com
+```
+
+* Good, because no wildcard certs (more secure).
+* Bad, because the number of SANs per cert is limited.
```

Comment:
The rate limit would also probably hit us here. Because we are also constantly replacing the same certificate in place, it could also be risky depending on the order things are done on an ACME server for example (if the server first revokes the previous certificate, it could create outages)